Add installer tests project

NikolaMilosavljevic opened the pull request:
Contributes to https://github.com/dotnet/runtime/issues/109848

This PR introduces test project for installer tests. It does not hook it up to build infra. There will be another PR soon that will introduce a job/step in `VMR Validation` stage that will run installer tests for both x64 and arm64 installers.

Originally the plan was to run these tests as part of the mainline build job, but we are running that job as cross-build on x64 host, which precludes us from testing arm64 installers.

Some notes about the PR contents:
- There is one RPM and one Debian test platform - this can be expanded as needed
- Debian installation uses `--force-all` to work around issues in https://github.com/dotnet/source-build/issues/4895 - the fix is in PR and this will be removed in the near future. I opted not to include that fix as a patch, but that is a possibility
- Consumers can specify if they want to test just Deb or Rpm or both
- `DockerHelper.cs` was borrowed from dotnet-docker repo: https://github.com/dotnet/dotnet-docker/blob/main/tests/Microsoft.DotNet.Docker.Tests/DockerfileHelper.cs. I have removed many unused methods
- Only Linux installer tests are being added today, Windows and MacOS installers will be considered in the future

NikolaMilosavljevic committed:
Add installer tests project

NikolaMilosavljevic commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+            }
+            else
+            {
+                Assert.Fail("Test summary not found");
```
Not true - `Assert.Fail` exists and is new in v2 2.5, see https://xunit.net/docs/comparisons

jkoritzinsky commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+
+    private readonly string[] RpmDistroImages =
+    [
+        "mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-40"
```
We don't ship the Fedora packages any more (and I'm actually in the process of removing the runtime-deps package in another PR). Can we instead use Azure Linux 3.0 or OpenSuSE Leap 15.6 here? Those are both RPM distros where we currently ship the packages.

Also, could we use the shipping "runtime-deps" images as our baseline instead of the buildtools-prereqs images? The shipping images better represent our customer scenarios.

NikolaMilosavljevic replied:
Thanks - will fix.

NikolaMilosavljevic replied:
A deps image would likely be this one `mcr.microsoft.com/dotnet/runtime-deps:9.0-azurelinux3.0` as we don't have 10.0 images yet. I don't think we want the OS base image like `mcr.microsoft.com/azurelinux/base/core:3.0`

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/054366b087810078c0e6e27f6f47edbbb5f765bd

jkoritzinsky commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+
+    private readonly string[] DebDistroImages =
+    [
+        "mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-24.04"
```
We don't ship the Ubuntu packages any more (and I'm actually in the process of removing the Ubuntu support from the runtime-deps package in another PR). Can we instead use Debian stable here? Debian is the only disto that we ship our Deb packages for now.

Also, could we use the shipping "runtime-deps" image as our baseline instead of the buildtools-prereqs image? The shipping images better represent our customer scenarios.

NikolaMilosavljevic replied:
Thanks - yeah makes sense, I'll update this.

NikolaMilosavljevic replied:
Deps image would likely be this one: `mcr.microsoft.com/dotnet/runtime-deps:9.0-bookworm-slim`

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/054366b087810078c0e6e27f6f47edbbb5f765bd

jkoritzinsky commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+        string matchPattern = "dotnet-runtime-deps-*.deb";
+        if (distroType == DistroType.Rpm)
+        {
+            string depsId = "fedora";
+            if (baseImage.Contains("fedora"))
+            {
+                depsId = "fedora";
+            }
+            else if (baseImage.Contains("centos"))
+            {
+                depsId = "centos";
+            }
+            else if (baseImage.Contains("rhel"))
+            {
+                depsId = "rhel";
+            }
+            else if (baseImage.Contains("opensuse"))
```
I'm removing these deps packages, so we don't need these checks here.

```suggestion
            string? depsId = null;
            if (baseImage.Contains("opensuse"
```

NikolaMilosavljevic replied:
Thanks, I'll incorporate this in an offline change, along other proposed changes.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/054366b087810078c0e6e27f6f47edbbb5f765bd

jkoritzinsky commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+    {
+        // For rpm enumerate RPM packages, excluding those that contain ".cm." in the name
+        List<string> rpmPackages = Directory.GetFiles(Config.AssetsDirectory, "*.rpm", SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("-cm.") && !Path.GetFileName(p).EndsWith("azl.rpm"))
```
Does this work if we were to test the Azure Linux 3.0 packages?

NikolaMilosavljevic replied:
Regular rpm packages have the exact same content as those with `azl.rpm` extension. Yes, it will work fine.

NikolaMilosavljevic replied:
`rpm` seems to only care if digests are ok. Will double-check this process with packages signed with regular Linux signatures.

NikolaMilosavljevic replied:
Yeah, `rpm` shows a warning when cert is not available, but happily installs the package. So, I think we can keep the existing code.

jkoritzinsky commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,386 @@
+        sb.Append("RUN");
+
+        // TODO: remove --force-all when aspnet package versioning issue have been resolved - https://github.com/dotnet/source-build/issues/4895
+        string packageInstallationCommand = distroType == DistroType.Deb ? "dpkg -i --force-all" : "rpm -i";
```
Could we construct these tests to point to a fake local package feed and install the packages via apt/tdnf/zypper instead of dpkg/rpm directly?

Doing so would let us better test actual end user scenarios

NikolaMilosavljevic replied:
That's something we could consider in the future. It's not as trivial to now add that to current tests, at least not as easy as adding a local NuGet store. It would be easy to reuse most of the code for those tests.

For now we're automating current manual testing of RPMs an DEBs, in the same way they are executed.

Utilizing correct installation order also verifies the expected package dependencies.

NikolaMilosavljevic committed:
Addressing PR comments

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+
+    private readonly string[] RpmDistroImages =
+    [
+        "mcr.microsoft.com/dotnet/runtime-deps:9.0-azurelinux3.0"
```
The version should be shared across RPM and Deb to ensure consistency and reduce maintenance.

I am afraid this is going to get stale quickly.

This should be using the 10-preview images out of nightly.

NikolaMilosavljevic replied:
Makes sense - will update.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+    private const string NetStandard21RpmPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.rpm";
+    private const string NetStandard21DebPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.deb";
+
+    private enum DistroType
```
```suggestion
    private enum PackageType
```

NikolaMilosavljevic replied:
Will fix - thanks!

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+    [Fact]
+    public void RunScenarioTestsForAllDistros()
+    {
+        if (Config.TestRpmPackages)
```
Ideally these would be separate testcases.  One of the primary reasons for this is that if one fails you would still get a read on the other.  Also having explicitly testcases helps make it more obvious what is being tested.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/commit/64fa1b8e6d800687ab49eae0b2a483e937a83e73

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+    private readonly string _tmpDir;
+    private readonly string _contextDir;
+
+    private readonly string[] RpmDistroImages =
```
Why is this an array?  What is the thinking on how this would expand?

NikolaMilosavljevic replied:
We will likely add more images. Having it as array makes the future modification simpler.

NikolaMilosavljevic replied:
I'm changing this to `InlineData` in a `Theory`.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/commit/64fa1b8e6d800687ab49eae0b2a483e937a83e73

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        {
+            try
+            {
+                OutputHelper.WriteLine($"Begin testing distro: {image}");
```
This message is a little strange to me because the purpose of the test is not to test the runtime-deps image.  It is testing the packages.

NikolaMilosavljevic replied:
Yeah. It is testing packages on that specific image - I'll clarify. It helps to have these status messages in the log.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+
+    private void InitializeContext()
+    {
+        // For rpm enumerate RPM packages, excluding those that contain ".cm." in the name
```
Why is this copying rpm files if only deb packages are being tested and vice versa for rpm only scenarios?  Why not encapsulate this in a package type specific initialize method?

NikolaMilosavljevic replied:
This will be modified when I split into two test cases, one for Deb and one for Rpm. Original goal was to create one shared context for all tests on all distros, rpm or deb based.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/a459cb9f457ca67b2f79ae11705fcb24b7314adb

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        DownloadFileAsync(NetStandard21DebPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21DebPackage))).Wait();
+
+        // Copy nuget packages
+        string nugetPackagesDir = Path.Combine(_contextDir, "packages");
```
Isn't this a sizeable copy?  Could a docker mount be utilized instead?  It would be more efficient.

NikolaMilosavljevic replied:
Good idea. Yes this would copy more than a GB of data.

NikolaMilosavljevic replied:
This copy is actually required, as we need to produce a flat layout of packages, for nuget restore use. Provided packages are likely in a tree-level structure, as is the case with `packages` dir in the `artifacts` dir.

It would be possible to keep the original directory structure and introduce many local sources in nuget.config, one source for each of the sub-directories that contain *.nuget packages. It would make code more complex, with limited performance benefits, about 1 second per test.

NikolaMilosavljevic replied:
I did try this approach and it worked when sharing a directory from my Windows host machine, but it failed in container build when sharing files on separate docker volume. This does complicate things quite a bit, as test container needs to mount all volumes that build container has.

I think we should proceed without this small improvement as it is not worth this extra effort.

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        }
+        finally
+        {
+            _dockerHelper.DeleteImage(tag);
```
Consider a runtimeconfig option to skip this.  I could see this as useful when debugging failures locally.

NikolaMilosavljevic replied:
Will do - thanks!

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/1f69d814a1c420251a8c18b94ada25608f2b2487

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+            {
+                output = e.Message;
+            }
+            Assert.Fail($"Build failed: {output}");
```
How do you know this failed during the build vs run?

NikolaMilosavljevic replied:
We don't - will refactor this.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/commit/64fa1b8e6d800687ab49eae0b2a483e937a83e73

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        sb.AppendLine("");
+        sb.AppendLine($"ENTRYPOINT [ \"dotnet\", \"{scenarioTestsBinary}\", \"--dotnet-root\", \"/usr/share/dotnet\" ]");
+
+        string dockerfile = Path.Combine(_contextDir, Path.GetRandomFileName());
```
I feel like debugging would be easier if the filename started with `Dockerfile` with a random suffix.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        if (distroType == DistroType.Rpm)
+        {
+            string? depsId = null;
+            if (baseImage.Contains("opensuse"))
```
Why is this not hardcoded to azurelinux since it appears to be the only exercisable codepath?

MichaelSimons replied:
Why not refactor at that time?  I don't like the thought of adding code that is not certain to be used, it is a form of technical debt.  We don't have runtime-deps images for these distros.  What is the ROI of adding them?

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+    private string GetContentPackage(string prefix, DistroType distroType)
+    {
+        string matchPattern = DistroType.Deb == distroType ? "*.deb" : "*.rpm";
+        string[] rpmFiles = Directory.GetFiles(_contextDir, prefix + matchPattern, SearchOption.AllDirectories)
```
These aren't necessarily rpm packages right?  They could be deb files as well?  If so the variable and exception message should be updated.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,374 @@
+        sb.AppendLine($"COPY packages packages");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy RPM packages");
```
These aren't necessarily RPM packages.  There are a few RPM references

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

NikolaMilosavljevic commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/Microsoft.DotNet.Installer.Tests.csproj:
```diff
@@ -0,0 +1,38 @@
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" />
```
Will take a look and update. We might not even need a dependency.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6

NikolaMilosavljevic committed:
Address more review comments

NikolaMilosavljevic committed:
Split the tests and use conditional theory

NikolaMilosavljevic commented:
With https://github.com/dotnet/sdk/pull/46927/commits/64fa1b8e6d800687ab49eae0b2a483e937a83e73, I've split the tests and switched to `ConditionalTheory`. I did have to use `repo` and `tag` as separate inline data, as xUnit prints only 50 characters of any inline data string. This can only be tweaked in xUnit v3.

I was getting the following, when using image as inline data:
`Passed Microsoft.DotNet.Installer.Tests.LinuxInstallerTests.DebTest(image: "mcr.microsoft.com/dotnet/nightly/runtime-deps:10.0"···) [6 m 10 s]`

Splitting into `repo` and `tag` produces a nicer output:
`Passed Microsoft.DotNet.Installer.Tests.LinuxInstallerTests.DebTest(repo: "mcr.microsoft.com/dotnet/nightly/runtime-deps", tag: "10.0-preview-trixie-slim") [6 m 14 s]`

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,350 @@
+    }
+
+    [ConditionalTheory(typeof(LinuxInstallerTests), nameof(IncludeRpmTests))]
+    [InlineData("mcr.microsoft.com/dotnet/nightly/runtime-deps", "10.0-preview-azurelinux3.0")]
```
Consider shared constants here to ensure consistency and reduce maintenance across the Rpm and Deb testcases.
```suggestion
    [InlineData(RuntimeDepsRepo, $"{RuntimeDepsVersion}-azurelinux3.0")]
```

NikolaMilosavljevic replied:
Yeah, that's good - will fix.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/1f69d814a1c420251a8c18b94ada25608f2b2487

NikolaMilosavljevic committed:
Few tweaks

NikolaMilosavljevic committed:
Refactor context initialization

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+        Deb
+    }
+
+    private ITestOutputHelper OutputHelper { get; set; }
```
Curious why a property is used here but fields for all other values?

NikolaMilosavljevic replied:
I did originally use a field, but changed to property to match the borrowed `DockerHelper` class. There is no need for that matching, of course. I'll change it back to field.

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+                if (clearElement != null)
+                {
+                    XElement newAddElement = new XElement("add",
+                    new XAttribute("key", "local-packages"),
```
```suggestion
                        new XAttribute("key", "local-packages"),
```

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+    private bool AnyTestFailures(string testResultSummary)
+    {
+        var parts = testResultSummary.Split(',')
+         .Select(part => part.Split(':').Select(p => p.Trim()).ToArray())
```
These should be fully indent IMO. It looks like they are only indented one space.

NikolaMilosavljevic replied:
Thanks - will fix.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/7dd918feef4c476fe34ed8ec3c51da831c3ad621

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+        string[] files = Directory.GetFiles(_contextDir, prefix + matchPattern, SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("dotnet-runtime-deps-"))
+            .ToArray();
+        if (files.Length == 0)
```
Curious if multiple are expected?

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+
+    private void DistroTest(string baseImage, PackageType packageType)
+    {
+        // Order of installation is important as we do not want to use "--nodeps"
```
Consider factoring out the logic to define the package list into a separate method.  IMO it hurts the readability of the method.

NikolaMilosavljevic replied:
Yes, this is something I considered earlier, makes sense to do it now.

NikolaMilosavljevic replied:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/7dd918feef4c476fe34ed8ec3c51da831c3ad621

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+        sb.AppendLine("# Install the installer packages and Microsoft.DotNet.ScenarioTests.SdkTemplateTests tool");
+        sb.Append("RUN");
+
+        // TODO: remove --force-all when aspnet package versioning issue have been resolved - https://github.com/dotnet/source-build/issues/4895
```
It looks like this has been resolved.  Can the workaround be removed now?

NikolaMilosavljevic replied:
Yes, this should be in VMR now - will confirm and update the code.

NikolaMilosavljevic replied:
We still need this workaround due to Debain DEPs package that requires `libgssapi-krb5-2`. The package is missing on `mcr.microsoft.com/dotnet/nightly/runtime-deps:10.0-preview-trixie-slim`

https://github.com/dotnet/dotnet-docker/issues/6271

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+
+        // Set entry point
+        sb.AppendLine("");
+        sb.AppendLine($"ENTRYPOINT [ \"dotnet\", \"{scenarioTestsBinary}\", \"--dotnet-root\", \"/usr/share/dotnet\" ]");
```
Curious why the entrypoint approach was used versus specifying via Docker run?

NikolaMilosavljevic replied:
No particular reason. Does `docker run` approach provide any additional benefits? I understand that entrypoint can be overridden when running this image, i.e. for debugging.

MichaelSimons replied:
I was mainly asking from a readability perspective of the DistroTest method - it is not obvious what tests are actually being run.

NikolaMilosavljevic replied:
Makes sense. Yeah, you're right. The way it is today, this command is hidden in dockerfile generation method. I'll update this in a follow up PR.

NikolaMilosavljevic replied:
I have a fix for this and will push the change soon.

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,373 @@
+            _dockerHelper.Build(tag, dockerfile: dockerfile, contextDir: _contextDir);
+            buildCompleted = true;
+            output = _dockerHelper.Run(tag, tag);
+
```
Would it make sense to capture the scenario-tests xml log file itself.  Would this help in diagnosing scenario-test failures?

NikolaMilosavljevic replied:
I don't think this is needed, as we dump the whole output, which would contain the complete failure log.

NikolaMilosavljevic committed:
Update src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

Co-authored-by: Michael Simons <msimons@microsoft.com>

NikolaMilosavljevic committed:
Address few more comments

NikolaMilosavljevic committed:
Specify test command in docker run command

NikolaMilosavljevic commented:
I need help merging this - it is hitting the unrelated test issues that are being worked on atm. @MichaelSimons

MichaelSimons commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,365 @@
+
+    private const string NetStandard21RpmPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.rpm";
+    private const string NetStandard21DebPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.deb";
+    private const string RuntimeDepsRepo = "mcr.microsoft.com/dotnet/nightly/runtime-deps";
```
Now that Preview 1 was released - this should be switched to "mcr.microsoft.com/dotnet/runtime-deps".  @NikolaMilosavljevic, can you include that change as part of hooking these tests up to a pipeline?  TIA

ViktorHofer commented on src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs:
```diff
@@ -0,0 +1,365 @@
+            File.Copy(Config.ScenarioTestsNuGetConfigPath, newNuGetConfig);
+            InsertLocalPackagesPathToNuGetConfig(newNuGetConfig, "/packages");
+
+            // Find the scenario-tests package and unpack it to the context dir, subfolder "scenario-tests"
+            string? scenarioTestsPackage = Directory.GetFiles(nugetPackagesDir, "Microsoft.DotNet.ScenarioTests.SdkTemplateTests*.nupkg", SearchOption.AllDirectories).FirstOrDefault();
```
I'm struggling to follow the execution flow here. Where is that package coming from?

ViktorHofer replied:
Does this project have a dependency on the scenario-tests.proj build?

NikolaMilosavljevic replied:
`Microsoft.DotNet.ScenarioTests.SdkTemplateTests*.nupkg` package is a product of the `scenario-tests` repo. I did mention this yesterday at the meeting as a need to have this repo built separately from testing it - as it is today. 

However, the plan is for installer tests to run during `VMR Validation` stage, and download all packages produced during VMR build. Scenario-tests package should be built, if not, that should be changed.

I will be testing the second part of installer-tests work soon, the pipeline hook-up.

NikolaMilosavljevic replied:
If the abovementioned plan is not possible, I'd add a dependency on scenario-tests repo project and have it built as part of installer testing.
